### PR TITLE
Update to UnRAR v6.2.10

### PR DESCRIPTION
**Pull request by micahsnyder**

### CODE REVIEW

File: libclamunrar/extract.cpp [hunk 1]
```diff
@@ -646,9 +646,9 @@ bool CmdExtract::ExtractCurrentFile(Archive &Arc,size_t HeaderSize,bool &Repeat)
       wcsncpyz(DestFileName,Cmd->DllDestName,ASIZE(DestFileName));
 #endif
 
-    if (ExtrFile && Command!='P' && !Cmd->Test && !Cmd->AbsoluteLinks &&
-        ConvertSymlinkPaths)
-      ExtrFile=LinksToDirs(DestFileName,Cmd->ExtrPath,LastCheckedSymlink);
+    // if (ExtrFile && Command!='P' && !Cmd->Test && !Cmd->AbsoluteLinks &&
```
Comment (shutton) [inline libclamunrar/extract.cpp:649]: Someone once told me, "don't commit commented-out code -- that's what SCM is for"

**Review by shutton — APPROVED** [review]
Thank goodness for "hide whitespace"

Seems fine. Probably no need to keep the old code around. If you do, I'd add a comment about what it *used* to do, and why you took it out.

File: libclamunrar/extract.cpp (hunk 1 above)
Comment (micahsnyder) [reply libclamunrar/extract.cpp:649]: Haha that's a good point.  I say the same thing sometimes. Okay, will just delete it.

Comment (micahsnyder) [reply libclamunrar/extract.cpp]: Fixed.

**Comment by micahsnyder** [comment]
I'm still trying to figure out why GitHub's Mac pipeline is failing to use C++11 (and failing the the build).  
Works fine on our Mac arm64 universal binary static-dependencies `.pkg` build, and also fine on our Mac x64 building just for x64...

**Comment by micahsnyder** [comment]
Silly me.  It was `unrar_iface` that was failing to build, because it includes `rar.hpp` which requires C++11 features.  Fixed with https://github.com/Cisco-Talos/clamav/pull/1008/commits/a8fdb2b3770b84caf19c2474bc98a3631f3a6a36

Now just to watch the test pipelines...

**Comment by micahsnyder** [comment]
Testing looking good after that last fix.

**Pull request merged**